rlp/raw_test.go

```diff
@@ -60,15 +60,43 @@ func TestCountValues(t *testing.T) {
 	}
 }
 
-func TestSplitTypes(t *testing.T) {
-	if _, _, err := SplitString(unhex("C100")); err != ErrExpectedString {
-		t.Errorf("SplitString returned %q, want %q", err, ErrExpectedString)
+func TestSplitString(t *testing.T) {
+	tests := []struct {
+		input string
+		err   error
+	}{
+		{"C0", ErrExpectedString},
+		{"C100", ErrExpectedString},
+		{"C3010203", ErrExpectedString},
+		{"C88363617483646F67", ErrExpectedString},
+		{"F8384C6F72656D20697073756D20646F6C6F722073697420616D65742C20636F6E7365637465747572206164697069736963696E6720656C6974", ErrExpectedString},
 	}
-	if _, _, err := SplitList(unhex("01")); err != ErrExpectedList {
-		t.Errorf("SplitString returned %q, want %q", err, ErrExpectedList)
+
+	for i, test := range tests {
+		if _, _, err := SplitString(unhex(test.input)); err != test.err {
+			t.Errorf("test %d: error mismatch: got %q, want %q", i, err, test.err)
```

Comment:
```suggestion
			t.Errorf("test %d: error mismatch: have %q, want %q", i, err, test.err)
```
For some reason, `have` and `want` are more go-idiomatic